src/VisualStudio/CSharp/Impl/Options/Formatting/CodeStylePage.cs

```diff
@@ -11,7 +19,502 @@ internal class CodeStylePage : AbstractOptionPage
+            // Null-checking preferences
+            editorconfig.AppendLine("# Null-checking preferences");
+            editorconfig.Append("csharp_style_throw_expression = ");
+            editorconfig.AppendLine(CSharpCodeStyleOptions_GenerateEditorconfig(optionSet, CodeStyleOptions.PreferThrowExpression));
+            editorconfig.Append("csharp_style_conditional_delegate_call = ");
+            editorconfig.AppendLine(CSharpCodeStyleOptions_GenerateEditorconfig(optionSet, CSharpCodeStyleOptions.PreferConditionalDelegateCall));
+
+            editorconfig.AppendLine();
+            // Modifier preferences
+            editorconfig.AppendLine("# Modifier preferences");
+            editorconfig.Append("csharp_preferred_modifier_order = ");
+            editorconfig.AppendLine(CSharpCodeStyleOptions_GenerateEditorconfig(optionSet, CSharpCodeStyleOptions.PreferredModifierOrder));
+
+            editorconfig.AppendLine();
+            // Expression-level preferences
+            editorconfig.AppendLine("# Expression-level preferences");
+            editorconfig.Append("csharp_prefer_braces = ");
+            editorconfig.AppendLine(CSharpCodeStyleOptions_GenerateEditorconfig(optionSet, CSharpCodeStyleOptions.PreferBraces));
+            editorconfig.Append("csharp_style_deconstructed_variable_declaration = ");
+            editorconfig.AppendLine(CSharpCodeStyleOptions_GenerateEditorconfig(optionSet, CodeStyleOptions.PreferDeconstructedVariableDeclaration));
+            editorconfig.Append("csharp_prefer_simple_default_expression = ");
+            editorconfig.AppendLine(CSharpCodeStyleOptions_GenerateEditorconfig(optionSet, CSharpCodeStyleOptions.PreferSimpleDefaultExpression));
+            editorconfig.Append("csharp_style_pattern_local_over_anonymous_function = ");
+            editorconfig.AppendLine(CSharpCodeStyleOptions_GenerateEditorconfig(optionSet, CSharpCodeStyleOptions.PreferLocalOverAnonymousFunction));
+            editorconfig.Append("csharp_style_inlined_variable_declaration = ");
+            editorconfig.AppendLine(CSharpCodeStyleOptions_GenerateEditorconfig(optionSet, CodeStyleOptions.PreferInlinedVariableDeclaration));
+
+            editorconfig.AppendLine();
+            // C# Formatting Rules
+            editorconfig.AppendLine("###############################");
+            editorconfig.AppendLine("# C# Formatting Rules         #");
+            editorconfig.AppendLine("###############################");
+
+            editorconfig.AppendLine();
+            // New line preferences
+            editorconfig.AppendLine("# New line preferences");
+            editorconfig.Append("csharp_new_line_before_open_brace = ");
+            editorconfig.AppendLine(CSharpNewLineBeforeOpenBrace_GenerateEditorconfig(optionSet));
+            editorconfig.Append("csharp_new_line_before_else = ");
+            editorconfig.AppendLine(optionSet.GetOption(CSharpFormattingOptions.NewLineForElse).ToString().ToLower());
```

Comment:
all the .ToString().ToLower() should be encapsulated in a helper function so we don't need to repeat it every time.